### PR TITLE
feat: add new web2 login tracking events

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: #2285

I've added all events belonging to the "web 2.0 log in" App Section (column D) from https://docs.google.com/spreadsheets/d/1JXw_pDC4QyqtcdYgxpMkbTVc24E2yz-qpwxvzelPfnY/edit#gid=465273655
That corresponds to rows 67 to 78.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

From https://deploy-preview-2310--regen-marketplace.netlify.app/, based on https://docs.google.com/spreadsheets/d/1JXw_pDC4QyqtcdYgxpMkbTVc24E2yz-qpwxvzelPfnY/edit#gid=465273655, test that all events from "web 2.0 log in" App Section are emitted.

You need to check the [**development** app on google analytics](https://analytics.google.com/analytics/web/?authuser=1#/p340668480/realtime/overview?params=_u..nav%3Dmaui&collectionId=life-cycle). 
**Also make sure you don't have any tracker blocker enabled.**

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/components/cards/ProjectCard/ProjectCard.tsx [hunk 1]
```diff
@@ -241,7 +241,7 @@ export function ProjectCard({
                       event.stopPropagation();
                       track &&
                         !isPrefinanceProject &&
-                        track<'buy1', Buy1Event>('buy1', {
```
Comment (blushi) [inline web-components/src/components/cards/ProjectCard/ProjectCard.tsx:244]: not directly related to this task, but here I'm just cleaning up the first generic param that was not really needed

**Comment by blushi** [comment]
@erikalogie see testing instructions

**Comment by erikalogie** [comment]
@blushi I'm not seeing the loginGoogle, enterCodeIncorrect, connectEmail events working in analytics. I can't test connectGoogle and disconnectGoogle because I'm getting this error: https://monosnap.com/file/qDRMrO3lTubqUJAQvc4PWUWIFHPoA0

**Comment by blushi** [comment]
@erikalogie I've updated some permission in google so we can use it for this deploy preview so you can now test `connectGoogle` and `disconnectGoogle`.

I can see the `loginGoogle` event, could you try again?

image

Same for `connectEmail`:

image


I've pushed some fix for `enterCodeIncorrect`.

**Comment by erikalogie** [comment]
Ok wonderful, enterCodeIncorrect now seems to be working correctly. 

I'm still not getting the loginGoogle, connectGoogle and disconnectGoogle events showing up in analytics: https://monosnap.com/file/58zd6PBEiN1WAmbxeYGxuxY7PBMRkW

**Review by aaronc — APPROVED** [review]

**Comment by blushi** [comment]
> @blushi I still don't see it in GA, but maybe I'm just testing wrong. Let's go ahead and merge this and I can retest on live after the release.

Ok interesting, I'll merge this but were you using https://deploy-preview-2310--regen-marketplace.netlify.app/ (not the dev.app.regen.network)?

**Comment by erikalogie** [comment]
> > @blushi I still don't see it in GA, but maybe I'm just testing wrong. Let's go ahead and merge this and I can retest on live after the release.
> 
> Ok interesting, I'll merge this but were you using https://deploy-preview-2310--regen-marketplace.netlify.app/ (not the dev.app.regen.network)?

Yes

**Pull request merged**